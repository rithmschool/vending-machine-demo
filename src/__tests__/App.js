import React from 'react'; // for JSX
import { mount } from 'enzyme'; // how to mount the component
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json'; // for snapshotting the component
import App from '../App'; // import the component itself

describe('<App />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <App />
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
