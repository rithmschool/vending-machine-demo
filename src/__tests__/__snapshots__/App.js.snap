// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        >
          <VendingMachine
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="VendingMachine"
              style={
                Object {
                  "backgroundImage": "url(machine.png)",
                }
              }
            >
              <Message>
                <div
                  className="Message"
                  id="messagey"
                >
                  <h1>
                    hello i am a vending machine. what would you like to eat?
                  </h1>
                </div>
              </Message>
              <Message>
                <div
                  className="Message"
                  id="messagey"
                >
                  <h1>
                    <Link
                      replace={false}
                      to="/soda"
                    >
                      <a
                        href="/soda"
                        onClick={[Function]}
                      >
                        soda
                      </a>
                    </Link>
                  </h1>
                  <h1>
                    <Link
                      replace={false}
                      to="/chips"
                    >
                      <a
                        href="/chips"
                        onClick={[Function]}
                      >
                        chips
                      </a>
                    </Link>
                  </h1>
                  <h1>
                    <Link
                      replace={false}
                      to="/sardines"
                    >
                      <a
                        href="/sardines"
                        onClick={[Function]}
                      >
                        fresh sardines
                      </a>
                    </Link>
                  </h1>
                </div>
              </Message>
            </div>
          </VendingMachine>
        </Route>
        <Route
          component={[Function]}
          exact={true}
          path="/soda"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/chips"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/sardines"
        />
      </div>
    </App>
  </Router>
</MemoryRouter>
`;
